CICE/patches/ice_domain.F90.patch

```diff
@@ -0,0 +1,71 @@
+diff --git a/cicecore/cicedyn/infrastructure/ice_domain.F90 b/cicecore/cicedyn/infrastructure/ice_domain.F90
+index 8b680f2..cf13fcc 100644
+--- a/cicecore/cicedyn/infrastructure/ice_domain.F90
++++ b/cicecore/cicedyn/infrastructure/ice_domain.F90
+@@ -88,7 +88,7 @@ module ice_domain
+        distribution_wght_file  ! file for distribution_wght=file
+ 
+     integer (int_kind) :: &
+-       nprocs                ! num of processors
++       nprocs, nprocs_x, nprocs_y                ! num of processors
+ 
+ !***********************************************************************
+ 
+@@ -101,7 +101,7 @@ subroutine init_domain_blocks
+ !  This routine reads in domain information and calls the routine
+ !  to set up the block decomposition.
+ 
+-   use ice_distribution, only: processor_shape
++   use ice_distribution, only: processor_shape, proc_decomposition
+    use ice_domain_size, only: ncat, nilyr, nslyr, max_blocks, &
+        nx_global, ny_global, block_size_x, block_size_y
+    use ice_fileunits, only: goto_nml
+@@ -204,6 +204,14 @@ subroutine init_domain_blocks
+ 
+    endif
+ 
++   if (nprocs .ne. -1) then
++      if (my_task == master_task) then
++         write(nu_diag,*) subname//' WARNING: nprocs is deprecated, please remove from namelist'
++      endif
++   endif
```

Comment:
You could move this to be inside the above `if` statement, so that there's no need to check again if `my_task == master_task`.